### DISCUSSION TITLE
Styling New OpenApi Documentation

### DISCUSSION THREAD

**Discussion by FadedPeppers** [GitbookIO/community#496] [4. Open API]
Hey! I was just curious how gitbook is styling their api requests on https://docs.gitbook.com/content-editor/blocks/openapi I would like to style my api documents similar to it, but unsure where to make the changes to reflect it. Specifically I really like having the examples to the right of the actual schema.

**Comment by addisonschultz** [comment]
@FadedPeppers great question! We're working on updating the way we render content in GitBook at the moment, including major updates to the OpenAPI block - as you can see! We're still hard at work on this, with a lot more improvements and features coming with. Stay tuned, you'll be able to see/use this feature soon!

**Reply by FadedPeppers** [reply]
Awesome! That is super exciting to hear! I know it may be hard to outright say, but any insights on how long it will be until those features are rolled out?

**Reply by martinezdylan** [reply]
Curious on this timeline. Given that API Method blocks have been deprecated for a bit, is there any chance we'll be able to leverage OpenAPI blocks to produce the UI you guys have published (ref. https://docs.gitbook.com/content-editor/blocks/openapi) anytime soon?

**Reply by addisonschultz** [reply]
@martinezdylan very soon! You should be able to see what it looks like for you, by adding ~open in the url. For instance, if your published site is at `https://open-source.gitbook.io/shot-scraper`, you can preview the updates coming soon at `https://open-source.gitbook.io/~open/shot-scraper`

let me know if that makes sense!

**Reply by martinezdylan** [reply]
Thanks so much for the update! I was able to apply this to our API docs and it rendered (everything but the fonts). Also, when opening the space it gave us an option to "preview" the changes so we were able to do a similar thing with `/~space/` prefix. Really looking forward to these changes going live, makes a HUGE difference for UX.

**Reply by addisonschultz** [reply]
@martinezdylan that is so awesome to hear! We've got a lot more coming after this release too. If you have some time, feel free to let us know you feedback in the form that pops up from the modal you see when clicking the "new" button in your space!

**Reply by martinezdylan** [reply]
Thanks @addisonschultz ! I've submitted some feedback which included a recommendation for controlling the content density & a small bug I found. I know its been a common question on timeline & you've mentioned very soon - but, the "New" button mentions March 1st being the upgrade (that it would be done automatically). Should this date be updated?

**Reply by addisonschultz** [reply]
@martinezdylan Thank you so much for the feedback! We've got it and will be adding it in as we continue to work on this 👍 

Good catch - March 1st was our aimed date, but we held back the release so we weren't releasing over the weekend! You should be seeing the update very shortly 🎉

**Reply by martinezdylan** [reply]
Hi @addisonschultz, could you share an updated timeline? I see there's a published ChangeLog from March 4th that mentions the changes already being live? ref. https://changelog.gitbook.com/march-2024/4-march-huge-improvements-to-your-published-documentation

Also, we no longer have the "New" button but are still left with the legacy OpenAPI blocks. Is it possible we're missing something?

**Reply by addisonschultz** [reply]
Hey @martinezdylan - great question! This release only affects published sites - so if you've published your site, there's a good chance you're already seeing the update! We've migrated most of the sites - but not all of them just yet. We will be finishing the migration hopefully this week 😄 

In regards to the editor - that is the next bit we're going to tackle - there will be some inconsistencies (with styling mainly) between the editor and the published sites until then

**Reply by martinezdylan** [reply]
I see. Our published page is updated but with the latest changes we're seeing some page not even being able to be rendered.. we were force pushed from API Method blocks so it's attempting to render the content as tables but this is what we're getting:

<img width="1512" alt="Screenshot 2024-03-12 at 11 25 50 AM" src="https://github.com/GitbookIO/community/assets/25836087/b4fe632a-2a5e-4b06-8314-02ad3c4f7263">

Also getting this on some pages:

<img width="1512" alt="Screenshot 2024-03-12 at 11 27 09 AM" src="https://github.com/GitbookIO/community/assets/25836087/88e7f542-c55c-46e1-8e0c-09540bae8d83">

Let me know the best place to make a formal troubleshooting request.

**Reply by addisonschultz** [reply]
Hey @martinezdylan - can you give a bit more context on when you're getting this? Is is only specific pages? If so can you please share the pages here? 

Also, do you know if others are getting the same?

**Reply by addisonschultz** [reply]
@martinezdylan also - do you maybe have a special setup, like multiple computers going through one single IP? Or have you been force re-loading your docs a few times?

**Reply by martinezdylan** [reply]
Correct, there are specific pages that with the new changes (legacy API method being converted to tables) I think caused a huge expanse in content which is resulting in a timeout - then from my IP i can't even access the docs altogether. Any machine on any network that navigates to that content will result the same.

No special setup, it's just a ton of content on the page & seems like you guys might be making individual requests for all of the content in sequence which is causing the rate limiting to trigger. The network trace shows the spammed GET requests - do you have a private drop I can share the HAR file?

**Reply by addisonschultz** [reply]
Can you email both me and my teammate @jpreynat? 

addison (at) gitbook.com
johan (at) gitbook.com

Thanks!

**Reply by martinezdylan** [reply]
Sweet. Should be in your inbox within the next few minutes. Thanks guys!
